src/lib/hoc/withAccessType.ts

```diff
@@ -51,19 +51,21 @@ export default function withAccessType(
       return loginRedirect;
     }
 
-    let user: PrivateProfile = JSON.parse(userCookie);
-    let userAccessLevel = user.accessType;
+    let user: PrivateProfile;
+    let userAccessLevel: UserAccessType;
 
-    // If no user cookie, try to re-generate one
-    if (!userCookie || !userAccessLevel) {
-      try {
+    try {
+      user = JSON.parse(userCookie);
+      userAccessLevel = user.accessType;
+      // If no user cookie, try to re-generate one
+      if (!userCookie || !userAccessLevel) {
```

Comment:
seeking to learn: what does this change?

Reply:
This is to prevent locking the user out if their user cookie contains malformed JSON for some reason

I made this change because my initial approach to update the user cookie after changing the profile was to set it to an empty string to force the server to regenerate it